include/gadget/sockets-map.h

```diff
@@ -172,23 +201,30 @@ gadget_socket_lookup(const struct __sk_buff *skb)
 	switch (key.proto) {
 	case IPPROTO_TCP:
 		if (skb->pkt_type == PACKET_HOST)
-			key.port = load_half(
-				skb, l4_off + offsetof(struct tcphdr, dest));
+			err = bpf_skb_load_bytes(
+				skb, l4_off + offsetof(struct tcphdr, dest),
+				&key.port, sizeof(key.port));
 		else
-			key.port = load_half(
-				skb, l4_off + offsetof(struct tcphdr, source));
+			err = bpf_skb_load_bytes(
+				skb, l4_off + offsetof(struct tcphdr, source),
+				&key.port, sizeof(key.port));
 		break;
 	case IPPROTO_UDP:
 		if (skb->pkt_type == PACKET_HOST)
-			key.port = load_half(
-				skb, l4_off + offsetof(struct udphdr, dest));
+			err = bpf_skb_load_bytes(
+				skb, l4_off + offsetof(struct udphdr, dest),
+				&key.port, sizeof(key.port));
 		else
-			key.port = load_half(
-				skb, l4_off + offsetof(struct udphdr, source));
+			err = bpf_skb_load_bytes(
+				skb, l4_off + offsetof(struct udphdr, source),
+				&key.port, sizeof(key.port));
 		break;
 	default:
 		return 0;
 	}
+	if (err < 0)
+		return 0;
```

Comment:
What about something like this?

```diff
diff --git include/gadget/sockets-map.h include/gadget/sockets-map.h
index 1e789f7d7..4ccbd226c 100644
--- include/gadget/sockets-map.h
+++ include/gadget/sockets-map.h
@@ -198,30 +198,26 @@ gadget_socket_lookup(const struct __sk_buff *skb)
                return 0;
        }

+       int off = l4_off;
+
        switch (key.proto) {
        case IPPROTO_TCP:
                if (skb->pkt_type == PACKET_HOST)
-                       err = bpf_skb_load_bytes(
-                               skb, l4_off + offsetof(struct tcphdr, dest),
-                               &key.port, sizeof(key.port));
+                       off += offsetof(struct tcphdr, dest);
                else
-                       err = bpf_skb_load_bytes(
-                               skb, l4_off + offsetof(struct tcphdr, source),
-                               &key.port, sizeof(key.port));
+                       off += offsetof(struct tcphdr, dest);
                break;
        case IPPROTO_UDP:
                if (skb->pkt_type == PACKET_HOST)
-                       err = bpf_skb_load_bytes(
-                               skb, l4_off + offsetof(struct udphdr, dest),
-                               &key.port, sizeof(key.port));
+                       off += offsetof(struct udphdr, dest);
                else
-                       err = bpf_skb_load_bytes(
-                       err = bpf_skb_load_bytes(
-                               skb, l4_off + offsetof(struct udphdr, source),
-                               &key.port, sizeof(key.port));
+                       off += offsetof(struct udphdr, source);
                break;
        default:
                return 0;
        }
+
+       err = bpf_skb_load_bytes(skb, off, &key.port, sizeof(key.port));
        if (err < 0)
                return 0;
        key.port = bpf_ntohs(key.port);
```

(It's even  possible to use the ternary operator to get a more concise code)